go/proof.go

```diff
@@ -40,20 +40,13 @@ var TendermintSpec = &ProofSpec{
 	},
 }
 
-// Verify does all checks to ensure this proof proves this key, value -> root
-// and matches the spec.
-func (p *ExistenceProof) Verify(spec *ProofSpec, root CommitmentRoot, key []byte, value []byte) error {
+// Verify does all checks to ensure this proof is a valid existence proof for this root
+// NOTE: In order to prove a value, we must call VerifyItem
+func (p *ExistenceProof) Verify(spec *ProofSpec, root CommitmentRoot) error {
```

Comment:
I understand that `Verify` will check the proof is valid (matches spec, and running the spec'd hash from leaf over all inner nodes will match the provided CommitmentRoot). But it doesn't say anything about what is being proven.

Reply:
Aye, should we even expose this function then? No IBC code should ever be calling `Verify` as defined like this.

Reply:
Maybe just add `key, value []byte` as extra parameters? Do we need a separation here?